azext_iot/tests/helpers.py

```diff
@@ -156,6 +159,127 @@ def get_agent_public_ip():
     return requests.head("https://www.wikipedia.org").headers["X-Client-IP"]
 
 
+def get_role_assignments(
+    scope: str,
+    assignee: str = None,
+    role: str = None,
+) -> json:
+    """
+    Get rbac permissions of resource.
+    """
+    role_flag = ""
+    assignee_flag = ""
+
+    if role:
+        role_flag = '--role "{}"'.format(role)
+
+    if assignee:
+        assignee_flag = '--assignee "{}"'.format(assignee)
+
+    return cli.invoke(
+        f'role assignment list --scope "{scope}" {role_flag} {assignee_flag}'
+    ).as_json()
+
+
+def assign_role_assignment(
+    role: str,
+    scope: str,
+    assignee: str,
+    max_tries=10,
+    wait=10
+):
+    """
+    Assign rbac permissions to resource.
+    """
+    output = None
+    tries = 0
+    while tries < max_tries:
+        role_assignments = get_role_assignments(scope=scope, role=role)
+        role_assignment_principal_ids = [assignment["principalId"] for assignment in role_assignments]
+        if assignee in role_assignment_principal_ids:
+            break
+        # else assign role to scope and check again
+        output = cli.invoke(
+            f'role assignment create --assignee "{assignee}" --role "{role}" --scope "{scope}"'
+        )
+        sleep(wait)
+        tries += 1
+    return output
```

Comment:
In the spirit of tightening up our test reliability - 
If this fails after `max_tries` - Should we throw an error or log a warning?
I see we return the output, which is then parsed in some locations. Theoretically, this command could:
- Succeed, but not show in `get_role_assignments` inside our polling window
- Fail, but require us to wait the entire polling window anyway

In the first situation (succeed but not complete), I expect the output would have all of the correct parameters (role ID, etc), but future calls will fail with perms / auth errors. In the second situation, it seems we'd see an error trying to parse the `assign_role_assignment` output, since I don't think we check for `success()` first.

Reply:
thanks! will add check for success()!